packages/shared/src/components/fields/PasswordField.tsx

```diff
@@ -64,8 +64,8 @@ export function PasswordField({
       valueChanged={onChange}
       leftIcon={<LockIcon size={IconSize.Small} />}
       hint={!!value && showStrength ? hint : props.hint}
-      validityChanged={setIsValid}
-      valid={isValid || !props.hint}
+      validityChanged={() => !props.hint && setIsValid}
```

Comment:
It still looks like we miss the passing of the actual value to setIsValid as the () is empty.
Not sure it makes a difference for areas where we use an internal hint actually.
As for instance the hint space password field (unused atm)